server/svix-server/src/core/security.rs

```diff
@@ -333,3 +336,57 @@ impl PartialEq for AsymmetricKey {
         self.0.as_slice() == other.0.as_slice()
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct Encryption(Option<Key>);
+
+impl Encryption {
+    const NONCE_SIZE: usize = 24;
+
+    pub fn new_noop() -> Self {
+        Self(None)
+    }
+
+    pub fn new(key: [u8; 32]) -> Self {
+        Self(Some(Key::from_slice(&key).to_owned()))
+    }
+
+    pub fn encrypt(&self, data: &[u8]) -> crate::error::Result<Vec<u8>> {
```

Comment:
I see we have e2e tests which probably suffice but this structure looks like it easily supports unit tests on `encrypt` and `decrypt` as well if we wanted to be thorough. Not prohibitive.

Reply:
Done.